src/open_r1/rewards.py

```diff
@@ -536,6 +536,34 @@ async def run_script(script: str, language: str, semaphore: asyncio.Semaphore) -
                 print(f"Error from E2B executor kill with sandbox ID {sandbox.sandbox_id} : {e}")
 
 
+def get_soft_overlong_punishment(max_completion_len, soft_punish_cache):
+    """
+    Reward function that penalizes overlong completions. It is used to penalize overlong completions,
+    but not to reward shorter completions. Reference: DAPO paper:https://huggingface.co/papers/2503.14476
+    Note: This function is response-level, not token-level.
```

Comment:
Elaborate a bit on this

```suggestion
    Note: In the DAPO paper, completion length is measured in token space, 
    whereas here it is measured in character space. While these units differ, 
    they are roughly proportional, meaning this function should yield a similar effect.
```